timbermill-local/src/main/java/com/datorama/oss/timbermill/ElasticsearchClient.java

```diff
@@ -311,12 +316,14 @@ public void index(Map<String, Task> tasksMap, String index) {
         Collection<Pair<Future, DbBulkRequest>> futuresRequests = createFuturesRequests(tasksMap, index);
 
         for (Pair<Future, DbBulkRequest> futureRequest : futuresRequests) {
-            try {
+			try {
 				futureRequest.getLeft().get();
-            } catch (InterruptedException | ExecutionException e) {
-                LOG.error("An error was thrown while indexing a batch", e);
-                failedRequests.offer(Pair.of(futureRequest.getRight(), 0));
-            }
+			} catch (InterruptedException e) {
+				LOG.error("An error was thrown while indexing a batch, going to retry", e);
+				sendDbBulkRequest(futureRequest.getRight());
+			} catch (ExecutionException e) {
+				LOG.error("An error was thrown while indexing a batch", e);
+			}
         }
```

Comment:
I'll explain the decision:
1) InterruptedException - as I understand it's hard to tell what is the context, then for safety simply call sendDbBulkRequest again.
2) ExecutionException - this error isn't related to the bulk method itself then I think it would repeat itself deterministically. Any way I can add a monitor to track this kind of failures

Reply:
2. go ahead

Reply:
add a monitor in DD that will notify me on such error